### PR TITLE
Make separate explicit API calls for evws_send_text and evws_send_binary (API breaking change)

### PR DESCRIPTION
**Pull request by widgetii**
This may fix https://github.com/libevent/libevent/issues/1483 issue

### CODE REVIEW

**Review by azat — APPROVED** [review]

**Comment by widgetii** [comment]
Sorry for delay, but I should've checked existing WS codebase and figured out that it's hard to maintain API break. So my proposal is still in place and by my opinion it's better to have separate `evws_send_text` and `evws_send_binary` calls

File: include/event2/ws.h [hunk 1]
```diff
@@ -34,10 +31,14 @@ EVENT2_EXPORT_SYMBOL
...
-/** Sends data over WebSocket connection */
+/** Sends text data over WebSocket connection */
+EVENT2_EXPORT_SYMBOL
+void evws_send_text(struct evws_connection *evws, const char *packet_str);
+
+/** Sends binary data over WebSocket connection */
 EVENT2_EXPORT_SYMBOL
-void evws_send(
-	struct evws_connection *evws, const char *packet_str, size_t str_len);
+void evws_send_binary(
```
Comment (azat) [inline include/event2/ws.h:40]: Can you also add a test for `ews_send_binary` ?

**Comment by azat** [comment]
Great, thank you! Let's also remove the merge commit from the history as well.

File: include/event2/ws.h (hunk 1 above)
Comment (widgetii) [reply include/event2/ws.h:43]: Done

**Comment by widgetii** [comment]
> Great, thank you! Let's also remove the merge commit from the history as well.

Tried several times with `git rebase -i` and friends, this simply doesn't work - it shows different commits unrelated to the PR

**Comment by azat** [comment]
>Tried several times with git rebase -i and friends, this simply doesn't work - it shows different commits unrelated to the PR

The problem is that merge commit contains conflict resolution, I've just squashed the commits.

**Pull request merged**